rfcs/text/0001-wildcard-data-type.md

```diff
@@ -48,46 +81,114 @@ Proposed type change:
 
-#### Paths and names
+Often ECS approaches semi-structured fields by breaking their values down into structured ones. These structured fields then enable better use of keyword fields' characteristics. For example, a URL can break down into its constituent parts: scheme, domain,  port, path, etc., and those parts can in turn be mapped to unique fields. However, not all fields are structured enough for this approach.
 
-Searching across various permutations of various file system and URL structures to find a particular file, path, or other resource. Leading wildcard searches on these fields may be necessary to users, but (as keywords) are known to produce very poor search performance
-(e.g. let's search for all instances of a known file without an unknown path: `file.path:*syslog.log`).
+In the security discipline, threat hunting searches and detection rules often rely on `grep`-like wildcard and regex patterns. Many other security platforms and SIEMs support wildcard and regex in this way, and it often can be confusing to security practitioners trying to adopt their detections and techniques to Elastic. These users require detailed pattern-matching operations that perform consistently well across large data sets. Wildcard is optimized for this type of need.
 
-Example data categories:
+A final interesting property of wildcard is its unlimited max character field size. Elasticsearch keyword field default to a max allowed string size of 256 characters which ECS ups to 1024 using `ignore_above`. Keyword can't be increased infinitely because Lucene has a hard set max limit of 32766 bytes for a single term. Due to how wildcard strings are indexed, they do not share this limitation. This makes wildcard the ideal data type for a very large string field (>32kB) that still needs to be indexed.
 
-* File paths
+The next sections details use cases which could benefit wildcard.
+
+### Paths
+
+Flexible nesting of a file path: `file.path:*\\Users\\*\\Temp\\*`
+Match under registry path: `registry.path:\\HKLM\\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\*\Debugger`
+A unique URL path: `url.full:https://api.example.com/account/*/foobar`
+
+The following are common categories
+
+* File and directory paths
 * URLs
+* Registry data
+
+### Names
+
+Similar to paths, different names components many need to be searched using one or more wildcard.
+
+Flexible host name searching: `host.name:prod-*-db*`
+Likewise, flexible searching of user names or accounts: `user.email:foo*@example.com`
```

Comment:
For usernames I think we'll want to stay on `keyword`, because usernames are typically a single word or two, and I'd be surprised that searching for parts of a username is a common thing.

Let's still keep the "Name" section however. I do think some of them may benefit. Upon further reflection, email may be a good fit (e.g. get me all of the gmail addresses: `email:*@gmail.com`), OS names as well, since they can be somewhat messy, and possibly domains names.

I think in stage 2 we should expand this section a bit, and try to capture some of the thinking on which names may benefit most.